roxie/udplib/udptrs.cpp

```diff
@@ -1025,6 +1025,25 @@ class CSendManager : implements ISendManager, public CInterface
 
         bool pushPermit(const UdpPermitToSendMsg &msg)
         {
+            auto updateTrackingIfMatches = [&msg](UdpPermitToSendMsg &m)
+            {
+                if (!m.matches(msg))
+                    return false;
+                m.seen = msg.seen; // Update the queue entry with the most recent tracking information
+                return true;
+            };
+
+            //First check to see if there is a matching permit in the queue.  If so update the tracking information and return.
+            if (send_queue.walk(updateTrackingIfMatches))
```

Comment:
Only concern would be if this is adding a lot of overhead. How big is the queue typically? This is called on a high priority thread...

Reply:
Also blocks the queue (with a critsec), so it's not just the size of the queue that might be a concern...

Reply:
I think it would be worth combining the update with the append - to avoid regaining the critical section.
I don't know how long the queue is normally.  It has a max capacity of 100, which is being filled up, but I think by duplicates.
I could add the new code on a switch so we could try with and without to compare (and see if my suspicion about duplicate items is true.)